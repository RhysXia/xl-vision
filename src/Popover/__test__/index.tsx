import { mount, ReactWrapper } from 'enzyme'
import * as React from 'react'
import Popover from '..'

describe('Popover', () => {
  let wrapper: ReactWrapper

  beforeAll(() => {
    jest.useRealTimers()
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('render', () => {
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    wrapper = mount(
      <Popover className='popup' visible={true} title='title' content='content'>
        <button type='button' className='btn'>
          button
        </button>
      </Popover>
    )

    expect(wrapper.render()).toMatchSnapshot()
  })
})
