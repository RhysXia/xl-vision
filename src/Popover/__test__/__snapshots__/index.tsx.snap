// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover render 1`] = `
Array [
  <button
    class="btn"
    type="button"
  >
    button
  </button>,
  <div
    data-popper-escaped=""
    data-popper-placement="top"
    data-popper-reference-hidden=""
    style="z-index: 1001; position: absolute; left: 0px; top: 0px; margin: 0px; padding-bottom: 10px; bottom: 0px; transform: translate(0px, 0px);"
  >
    <div
      class="xl-popover popup"
      data-placement="top"
      style="transform-origin: 0px 100%;"
    >
      <div
        class="xl-popover__arrow"
        data-popper-arrow=""
        style="position: absolute; left: 0px; transform: translate(0px, 0px);"
      />
      <div
        class="xl-popover__inner"
      >
        <div
          class="xl-popover__title"
        >
          title
        </div>
        <div
          class="xl-popover__content"
        >
          content
        </div>
      </div>
    </div>
  </div>,
]
`;
